user-service/src/main/java/uk/gov/gchq/palisade/service/user/web/UserRestController.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.user.web;
+
+import akka.Done;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import uk.gov.gchq.palisade.service.user.model.Token;
+import uk.gov.gchq.palisade.service.user.model.UserRequest;
+import uk.gov.gchq.palisade.service.user.stream.ConsumerTopicConfiguration;
+import uk.gov.gchq.palisade.service.user.stream.ProducerTopicConfiguration.Topic;
+
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping(path = "/api")
+public class UserRestController {
```

Comment:
There should be no Akka or Kafka stuff in here, it should be hidden behind a service. Refactor please.

Reply:
I've had to created an AsyncUserServiceProxy (this does not implement the UserService interface) in order to be able to deal with async and caching. Would you prefer this was moved to that new class or have another UserService implementation?

Reply:
Moved Kafka/Akka to KafkaProducerService class